From 61acb451878a41121281a4eafcee7601056fe37d Mon Sep 17 00:00:00 2001
From: Kotresh HR <khiremat@redhat.com>
Date: Mon, 21 Mar 2016 12:01:47 +0530
Subject: [PATCH 154/158] features/marker: Fix dict_get errors when key is NULL

Backport of:
master:
>BUG: 1319581
>Reviewed-on: http://review.gluster.org/13793

release-3.7:
>BUG: 1332074
>Reviewed-on: http://review.gluster.org/14144

BUG: 1335505
Change-Id: I73f6443a24518008869048b85602ff019009e420
Signed-off-by: Kotresh HR <khiremat@redhat.com>
Reviewed-on: https://code.engineering.redhat.com/gerrit/74369
Reviewed-by: Venky Shankar <vshankar@redhat.com>
Tested-by: Venky Shankar <vshankar@redhat.com>
---
 libglusterfs/src/glusterfs.h         |    1 +
 xlators/features/marker/src/marker.c |    6 +++---
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/libglusterfs/src/glusterfs.h b/libglusterfs/src/glusterfs.h
index 34c0361..c8c4590 100644
--- a/libglusterfs/src/glusterfs.h
+++ b/libglusterfs/src/glusterfs.h
@@ -159,6 +159,7 @@
 #define GLUSTERFS_INODELK_DOM_COUNT "glusterfs.inodelk-dom-count"
 #define GFID_TO_PATH_KEY "glusterfs.gfid2path"
 #define GF_XATTR_STIME_PATTERN "trusted.glusterfs.*.stime"
+#define GF_XATTR_XTIME_PATTERN "trusted.glusterfs.*.xtime"
 #define GF_XATTR_TRIGGER_SYNC "glusterfs.geo-rep.trigger-sync"
 
 /* quota xattrs */
diff --git a/xlators/features/marker/src/marker.c b/xlators/features/marker/src/marker.c
index 10bf531..6a1952c 100644
--- a/xlators/features/marker/src/marker.c
+++ b/xlators/features/marker/src/marker.c
@@ -424,10 +424,10 @@ marker_filter_gsyncd_xattrs (call_frame_t *frame,
 
         priv = this->private;
         GF_ASSERT (priv);
+        GF_ASSERT (frame);
 
-        if (frame->root->pid != GF_CLIENT_PID_GSYNCD &&
-            dict_get(xattrs, priv->marker_xattr)) {
-                dict_del (xattrs, priv->marker_xattr);
+        if (xattrs && frame->root->pid != GF_CLIENT_PID_GSYNCD) {
+                GF_REMOVE_INTERNAL_XATTR (GF_XATTR_XTIME_PATTERN, xattrs);
         }
         return;
 }
-- 
1.7.1

